<launch>
  <arg name="input" default="/rs_d435/aligned_depth_to_color/image_raw" />
  <arg name="rgb_image_topic" default="/rs_d435/color/image_rect_color" />
  <arg name="depth_image_topic" default="/rs_d435/aligned_depth_to_color/image_raw" />
  <arg name="camera_info_topic" default="/rs_d435/aligned_depth_to_color/camera_info" />
  <arg name="debug_view" default="true"/>
  <arg name="frame_id" default="target_object"/>
  <arg name="gui" default="true"/>

  <node pkg="nodelet" type="nodelet" name="rectangle_detection_nodelet" args="manager" output="log"/>

  <node pkg="nodelet" type="nodelet" name="depth_distance_filter"
        args="load aerial_robot_perception/DepthDistanceFilter rectangle_detection_nodelet" output="screen">
    <remap from="~input" to="$(arg input)" />

    <param name="min_depth" value="0.3"/>
    <param name="max_depth" value="1.9"/>
    <param name="use_distance_from_ground" value="true"/>
    <param name="distance_from_ground" value="0.1"/>
  </node>

  <node name="opening" pkg="nodelet" type="nodelet" args="load jsk_perception/Opening rectangle_detection_nodelet">
    <remap from="~input" to="/depth_distance_filter/output"/>
    <rosparam>
      size: 2
      iterations: 2
    </rosparam>
  </node>

  <node pkg="nodelet" type="nodelet" name="rectangle_detection"
        args="load aerial_robot_perception/RectangleDetection rectangle_detection_nodelet" output="screen">
    <remap from="~input" to="/opening/output" />
    <remap from="~rgb_img" to="$(arg rgb_image_topic)" />
    <remap from="~cam_info" to="$(arg camera_info_topic)" />
    <param name="debug_view" value="$(arg debug_view)"/>
    <param name="frame_id" value="$(arg frame_id)"/>
    <param name="target_object_area_margin" value="0.02"/>
  </node>

</launch>
